core/src/main/java/io/grpc/InternalServiceProviders.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2017, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.util.List;
+
+@Internal
+public final class InternalServiceProviders {
+  private InternalServiceProviders() {
+  }
+
+  /**
+   * Accessor for method.
+   */
+  public static <T> T load(
```

Comment:
load, loadAll, isAndroid, and PriorityAccessor don't seem like they need to be exposed.

I'd sort of prefer this remained an implementation detail and so we don't expose getCandidatesViaServiceLoader and getCandidatesViaHardCoded as well (putting those methods on each provider instead), but I can live with this.